pymeasure/display/manager.py

```diff
@@ -260,7 +261,13 @@ def _finish(self):
         experiment.browser_item.setProgress(100.)
         for curve in experiment.curve_list:
             if curve:
-                curve.update_data()
```

Comment:
I'm not sure if this change is indeed needed, if I checked correctly, the curve_list only contains ResultsCurves, and not a list of ResultsCurves.

(Additionally, if it would, I think it would be cleaner if that list would be unpacked into the curve_list earlier on, to keep things simple, but that seems to already happen in the `new_experiment` method of the ManagedWindowBase class.)

Reply:
At the moment this is needed, because the default implementation assumes each experiment may have a list of widgets and each of those widgets has one curve. `DockWidget` is one widget with multiple curves.

I think a cleaner implementation is needed but that may need a wider look the relationship between widgets and curves.

Reply:
hmmm yes... For some reason I thought I tried reverting this and seeing it work, but indeed this is now needed..  I think this shows problems of the underlying code; see my next review and PR to your branch  https://github.com/mcdo0486/pymeasure/pull/13.